### PR TITLE
Table: fix data is null and rowKey or defaultExpandAll error.

### LINKED ISSUE #16301: [Bug Report] Table to toggleRowExpansion Method Console Error Reporting
### Element UI version
2.10.0

### OS/Browsers version
windows10 / chrome 71

### Vue version
2.6.10

### Reproduction Link
https://codepen.io/PolanZ/pen/JQMGbv

### Steps to reproduce
当点击没有子级的行，toggleRowExpansion报错了，源码判断有错
PS：获取表格数据不对，null

### What is Expected?
虽然不影响，但是控制报错，就会被监听到，
希望完善好判断吧

### What is actually happening?
控制台出错
 Cannot read property 'expanded' of undefined

### PR DESCRIPTION
**Pull request by PolanZ**
### Table
- fix data is null and rowKey or defaultExpandAll error.
- fix toggleTreeExpansion method lack of rigorous judgement.

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

- Closes #16301

### CODE REVIEW

**Comment by island205** [comment]
related #16304

File: packages/table/src/store/expand.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ export default {
   methods: {
     updateExpandRows() {
       const { data = [], rowKey, defaultExpandAll, expandRows } = this.states;
+      if (!data) return;
```
Comment (ziyoung) [inline packages/table/src/store/expand.js]: 为什么加这一行呢？

File: packages/table/src/store/tree.js [hunk 2]
```diff
@@ -148,8 +148,9 @@ export default {
       const { rowKey, treeData } = this.states;
       const id = getRowIdentity(row, rowKey);
       const data = id && treeData[id];
+      if (!data) return;
       const oldExpanded = treeData[id].expanded;
-      if (id && data && ('expanded' in data)) {
+      if (id && ('expanded' in data)) {
```
Comment (ziyoung) [inline packages/table/src/store/tree.js]: 代码可以优化一下
```diff
- const oldExpanded = treeData[id].expanded;
+ const oldExpanded = data.expanded;
```
并且这一行还可以移到 if 语句中。

File: packages/table/src/store/expand.js (hunk 1 above)
Comment (PolanZ) [reply packages/table/src/store/expand.js]: 若是data: null就会报错哦

**Pull request closed without merging**